cypress/plugins/index.js

```diff
@@ -1,17 +1,82 @@
-// ***********************************************************
-// This example plugins/index.js can be used to load plugins
-//
-// You can change the location of this file or turn off loading
-// the plugins file with the 'pluginsFile' configuration option.
-//
-// You can read more here:
-// https://on.cypress.io/plugins-guide
-// ***********************************************************
-
-// This function is called when a project is opened or re-opened (e.g. due to
-// the project's config changing)
-/* eslint-disable no-unused-vars */
-module.exports = (on, config) => {
-  // `on` is used to hook into various events Cypress emits
-  // `config` is the resolved Cypress config
+const getChallenge = require('../../curriculum/getChallenges');
+const env = require('../../config/env.json');
+
+function createPath(curriculum, superblock, block) {
+  let challengePaths = [];
+  console.log('this is called but it does nothing');
+  let superBlockPath = superblock;
+  let blockPath = block;
+  let challengeArr = curriculum[superblock]['blocks'][block]['challenges'];
+
+  challengeArr.forEach(challengePath => {
+    challengePaths.push(
+      `/learn/${superBlockPath}/${blockPath}/${challengePath['dashedName']}`
+    );
+  });
+
+  return challengePaths;
+}
+
+module.exports = on => {
+  on('task', {
+    // return an array of all challenge paths
+    //  superBlock, block,  this are unused variables in the condition Husky doesn't like that
+    getCurriculum({ lang }) {
+      let curriculum;
+
+      if (!curriculum) {
+        curriculum = getChallenge.getChallengesForLang(lang);
+      }
+
+      return curriculum;
+    },
+
+    scopeCurriculum({ curriculum, superblock, block, challenge }) {
+      let challengePaths = [];
+
+      if (superblock && block === null) {
+        // Get blocks in superblock
+        block = Object.keys(curriculum[superblock]['blocks']);
+
+        // Get challenges in block
+
+        block.forEach(blocks => {
+          const upcommingChange =
+            curriculum[superblock]['blocks'][blocks]['meta'][
+              'isUpcomingChange'
+            ];
+
+          // Check if block is upcoming change
+          if (!upcommingChange) {
+            challengePaths = createPath(curriculum, superblock, blocks);
+          } else if (upcommingChange && env.showUpcomingChanges) {
+            challengePaths = createPath(curriculum, superblock, blocks);
+          }
```

Comment:
```suggestion
          if (!upcomingChange || env.showUpcomingChanges) {
            challengePaths = createPath(curriculum, superblock, blocks);
          }
```
Since it's the same function call each time, we can simplify here.